solver.rb

```diff
@@ -0,0 +1,50 @@
+require './solver'
+
+describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    it 'returns 1 when the input is 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns the correct factorial for positive integers' do
+      expect(solver.factorial(5)).to eq(120)
+      expect(solver.factorial(10)).to eq(3_628_800)
+    end
+
+    it 'raises an exception for negative integers' do
+      expect { solver.factorial(-1) }.to raise_error(ArgumentError)
+      expect { solver.factorial(-5) }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#reverse' do
+    it 'returns the word reversed' do
+      expect(solver.reverse('hello')).to eq('olleh')
+      expect(solver.reverse('world')).to eq('dlrow')
+    end
+  end
+
+  describe '#fizzbuzz' do
+    it 'returns "fizz" when the number is divisible by 3' do
+      expect(solver.fizzbuzz(3)).to eq('fizz')
+      expect(solver.fizzbuzz(9)).to eq('fizz')
+    end
+
+    it 'returns "buzz" when the number is divisible by 5' do
+      expect(solver.fizzbuzz(5)).to eq('buzz')
+      expect(solver.fizzbuzz(20)).to eq('buzz')
+    end
+
+    it 'returns "fizzbuzz" when the number is divisible by both 3 and 5' do
+      expect(solver.fizzbuzz(15)).to eq('fizzbuzz')
+      expect(solver.fizzbuzz(30)).to eq('fizzbuzz')
+    end
+
+    it 'returns the number as a string for any other case' do
+      expect(solver.fizzbuzz(7)).to eq('7')
+      expect(solver.fizzbuzz(8)).to eq('8')
+    end
+  end
+end
```

Comment:
- [ Required ] I think it would be better if you add a class called `Solver` which has the following methods `#factorial, #reverse`, and `#fizzbuzz` as requested by the previous reviewer. Currently, this project does not have a class called `Solver`. I would like to suggest that you look at the example code below as an example of how to achieve this.

## Example code
```ruby
class Solver
  def reverse(str)
    str.reverse
  end

  def fizzbuzz(number)
    if (number % 3).zero? && (number % 5).zero?
      'fizzbuzz'
    elsif (number % 3).zero?
      'fizz'
    elsif (number % 5).zero?
      'buzz'
    else
      number.to_s
    end
  end
end
```

## Project requirements
![Screenshot from 2023-08-04 17-07-28](https://github.com/Reggeditt/ruby-tdd/assets/16392046/1ff283ca-a676-4eeb-b6ee-b325dae5029a)

## Error in your project
![Screenshot from 2023-08-04 17-17-28](https://github.com/Reggeditt/ruby-tdd/assets/16392046/43a0e494-ccf9-442f-8149-60c60dc6e5c1)

Reply:
fixed